easybuild/easyblocks/o/openfoam.py

```diff
@@ -83,24 +83,38 @@ def extract_step(self):
         if not os.path.exists(openfoam_installdir):
             self.log.warning("Creating expected directory %s, and moving everything there" % openfoam_installdir)
             try:
-                mkdir(openfoam_installdir)
-                for fil in os.listdir(self.installdir):
-                    if fil != self.openfoamdir:
-                        source = os.path.join(self.installdir, fil)
-                        target = os.path.join(openfoam_installdir, fil)
-                        self.log.debug("Moving %s to %s" % (source, target))
-                        shutil.move(source, target)
-                os.chdir(openfoam_installdir)
+                contents_installdir = os.listdir(self.installdir)
+                # it's one directory but has a wrong name
+                if len(contents_installdir) == 1 and os.path.isdir(os.path.join(self.installdir, contents_installdir[0])):
+                    source = os.path.join(self.installdir, contents_installdir[0])
+                    target = os.path.join(self.installdir, self.openfoamdir)
+                    self.log.debug("Moving %s to %s" % (source, target))
+                    os.rename(source, target)
+                else:
+                    mkdir(openfoam_installdir)
+                    for fil in contents_installdir:
+                        if fil != self.openfoamdir:
+                            source = os.path.join(self.installdir, fil)
+                            target = os.path.join(openfoam_installdir, fil)
+                            self.log.debug("Moving %s to %s" % (source, target))
+                            shutil.move(source, target)
+                    os.chdir(openfoam_installdir)
             except OSError, err:
                 raise EasyBuildError("Failed to move all files to %s: %s", openfoam_installdir, err)
 
+    def patch_step(self):
+        """Adjust beginpath and start_dir to correct openfoam dir"""
+        beginpath = os.path.join(self.installdir, self.openfoamdir)
+        self.cfg['start_dir'] = beginpath
+        super(EB_OpenFOAM, self).patch_step(beginpath=beginpath)
```

Comment:
why is this required? are you sure this is not going to break existing OpenFOAM easyconfigs?

testing will tell...